### PR TITLE
Make more controllers from existing endpoints

### PR DESCRIPTION
**Pull request by Pierstoval**

### CODE REVIEW

**Comment by cconard96** [comment]
Is there some extra setup these PRs need? I'm getting 404s for everything that gets converted including the two APIs from the previous PR still.

**Comment by cedric-anne** [comment]
> Is there some extra setup these PRs need? I'm getting 404s for everything that gets converted including the two APIs from the previous PR still.

Can you try to remove the `files/_cache/symfony` directory to see if problem persist ? Do you have the `GLPI_ENVIRONMENT_TYPE` constant defined to `development` ?

**Comment by cconard96** [comment]
It seems like it was a cache issue and had to be manually deleted. Is it intended that this part isn't removed with the `cache:clear` command though?

**Comment by Pierstoval** [comment]
If `GLPI_ENVIRONMENT_TYPE` is set to `development`, cache should be refreshed on every HTTP request if config files were updated.

Will work on the `cache:clear` command though, to make sure to remove it 👍

**Comment by Pierstoval** [comment]
I added a call to Symfony `cache:clear` command.

I don't like how it's done, however this is the best way to support Kernel-based envs as well as non-Kernel envs (like testing or custom usage).

**Comment by cedric-anne** [comment]
> My local_define.php:
> 
> ```
> <?php
> 
> if (!defined('GLPI_MARKETPLACE_ENABLE')) {
>    define('GLPI_MARKETPLACE_ENABLE', 1);
> }
> 
> if (!defined('SKIP_UPDATES')) {
>     //define('SKIP_UPDATES', true);
> }
> 
> if (!defined('GLPI_ENVIRONMENT_TYPE')) {
>     define('GLPI_ENVIRONMENT_TYPE', 'development');
> }
> 
> if (!defined('GLPI_CENTRAL_WARNINGS')) {
>     define('GLPI_CENTRAL_WARNINGS', '0');
> }
> ```
> 
> The environment should be `development` for me. The info in Setup > General > System also shows it is `development`.

With this method, the `GLPI_ENVIRONMENT_TYPE` constant is defined after the call to the Kernel constructor. We have to find a solution for this. Maybe we should only support the definition of the `GLPI_ENVIRONMENT_TYPE` in the system/web server env variables.

**Comment by Pierstoval** [comment]
> With this method, the `GLPI_ENVIRONMENT_TYPE` constant is defined after the call to the Kernel constructor

I think we should not support a case where something is defined after the Kernel is created. This brings too many edge-cases, and will definitely make it harder to standardize the configuration into one service 😕

I'd suggest we move to env vars, with a `.env` file.
Thanks to Symfony's env var processors, we can import PHP or JSON files in our config in case there is no other alternative.
And I am certain we can also create a simple migration script that can convert a local config into a local env file, like `.env.local`, or Symfony's best choice with `.env.local.php` (which is faster to read)

**Comment by cedric-anne** [comment]
> > With this method, the `GLPI_ENVIRONMENT_TYPE` constant is defined after the call to the Kernel constructor
> 
> I think we should not support a case where something is defined after the Kernel is created. This brings too many edge-cases, and will definitely make it harder to standardize the configuration into one service 😕
> 
> I'd suggest we move to env vars, with a `.env` file. Thanks to Symfony's env var processors, we can import PHP or JSON files in our config in case there is no other alternative. And I am certain we can also create a simple migration script that can convert a local config into a local env file, like `.env.local`, or Symfony's best choice with `.env.local.php` (which is faster to read)

The `.env` support is another task, and is not yet planned. I will propose a small patch to handle this case to be able to merge the current work in the main branch in the next few weeks.

File: src/Glpi/Cache/CacheManager.php [hunk 1]
```diff
@@ -561,4 +571,23 @@ public static function getAvailableAdapters(): array
             self::SCHEME_REDISS     => __('Redis (TLS)'),
         ];
     }
+
+    private function clearSymfonyCache(): void
+    {
+        /** Kernel|null */
```
Comment (cedric-anne) [inline src/Glpi/Cache/CacheManager.php:577]:
```suggestion
        /** @var Kernel|null $kernel */
```

File: src/Glpi/Controller/IndexController.php [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    private function call(): void
+    {
+        // Check PHP version not to have trouble
+        // Need to be the very fist step before any include
+        if (
+            \version_compare(PHP_VERSION, '8.2.0', '<') ||
+            \version_compare(PHP_VERSION, '8.3.999', '>')
+        ) {
+            die('PHP version must be between 8.2 and 8.3.');
+        }
```
[the remark this answers is not in the document]
Comment (Pierstoval) [reply src/Glpi/Controller/IndexController.php:77]: Indeed, will do 👍

**Comment by Pierstoval** [comment]
All green, should be mergeable now!

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**